Create Readme and add MIT License (#1)
joaovds/gotterChat-backend: README.md

<div align="center">
  <img src="https://github.com/hmartiins/chat/assets/51277667/fca09cd1-d66a-4843-a6f4-cbbc1bfd738f" width="200" />

  <h1>Gotter Chat</h1>
  <p>💬 Application for exchanging messages in real time 📱</p>
</div>

## Project Overview

> This project is a backend implementation in Go for a real-time chat system using Socket.IO. The primary goal is to create a scalable and efficient server that facilitates communication between multiple clients in real-time.

Reviewer: What do you think of us using the term "WebSocket" instead of "Socket.IO"? Socket.IO, I believe, refers to a javascript library, and not to the type of communication we are going to use

```suggestion
This project is a backend implementation in Go for a real-time chat system using WebSocket. The primary goal is to create a scalable and efficient server that facilitates communication between multiple clients in real-time.
```

### Features

- **Real-time Communication**: Utilizing Socket.IO for bidirectional communication between the server and clients, enabling instant message updates.
- **Scalability**: Designed to handle a large number of concurrent connections efficiently, ensuring smooth communication even with a high volume of users.
- **User Authentication**: Implement a secure authentication mechanism to validate users and manage access to chat features.
- **Message Persistence**: Optionally, store chat messages in a database to provide message history and continuity across user sessions.
- **Error Handling**: Implement robust error handling to gracefully manage unexpected scenarios, ensuring the stability of the chat server.
- **Logging**: Utilize logging to keep track of important events and errors for monitoring and debugging purposes.

## Getting Started

### Prerequisites

- [Go](https://golang.org/) installed on your machine.

> - [Socket.IO Go Library](https://github.com/googollee/go-socket.io) installed.

Reviewer: I think you can remove that line. All the libs that the project depends on, Go's own package manager will take care of. Instead, perhaps you could put Docker as a dependency, as we will soon add a Dockerfile/docker-compose

### Installation

1. Clone the repository:

    ```bash

>     git clone https://github.com/your-username/chat-backend-go

Reviewer: ```suggestion
    git clone https://github.com/joaovds/gotterChat-backend
```
I will change the name of the repo to; gotterChat-backend

    ```

2. Navigate to the project directory:

    ```bash

>     cd chat

Reviewer: ```suggestion
    cd gotterChat-backend
```

    ```

> 

Reviewer: add the .env configuration session before running the project. Using "cp .env.example .env"

3. Install dependencies:

    ```bash

>     go mod download

Reviewer: ```suggestion
    go mod tidy
```

    ```

4. Run the server:

    ```bash

>     go run main.go

Reviewer: ```suggestion
    go run cmd/chat/main.go
```

    ```

## Configuration

> - Modify the `config.toml` file to customize the server settings, such as port, database connection details, and authentication configurations.

Reviewer: .env

<div align="center">
  <img src="https://github.com/hmartiins/chat/assets/51277667/fca09cd1-d66a-4843-a6f4-cbbc1bfd738f" width="200" />

  <h1>Gotter Chat</h1>
  <p>💬 Application for exchanging messages in real time 📱</p>
</div>

## Project Overview

This project is a backend implementation in Go for a real-time chat system using WebSocket. The primary goal is to create a scalable and efficient server that facilitates communication between multiple clients in real-time.

### Features

> - **Real-time Communication**: Utilizing Socket.IO for bidirectional communication between the server and clients, enabling instant message updates.

Reviewer: ```suggestion
- **Real-time Communication**: Utilizing WebSocket for bidirectional communication between the server and clients, enabling instant message updates.
```
